src/Symfony/Component/Messenger/Tests/Transport/Doctrine/DoctrineReceiverTest.php

```diff
@@ -28,14 +31,26 @@ public function testItReturnsTheDecodedMessageToTheHandler()
     {
         $serializer = $this->createSerializer();
 
-        $doctrineEnvelop = $this->createDoctrineEnvelope();
+        $doctrineEnvelope = $this->createDoctrineEnvelope();
         $connection = $this->getMockBuilder(Connection::class)->disableOriginalConstructor()->getMock();
-        $connection->method('get')->willReturn($doctrineEnvelop);
+        $connection->method('get')->willReturn($doctrineEnvelope);
 
         $receiver = new DoctrineReceiver($connection, $serializer);
         $actualEnvelopes = iterator_to_array($receiver->get());
         $this->assertCount(1, $actualEnvelopes);
+        /** @var Envelope $actualEnvelope */
+        $actualEnvelope = $actualEnvelopes[0];
         $this->assertEquals(new DummyMessage('Hi'), $actualEnvelopes[0]->getMessage());
+
+        /** @var DoctrineReceivedStamp $doctrineReceivedStamp */
+        $doctrineReceivedStamp = $actualEnvelope->last(DoctrineReceivedStamp::class);
+        $this->assertNotNull($doctrineReceivedStamp);
+        $this->assertSame('1', $doctrineReceivedStamp->getId());
+
+        /** @var TransportMessageIdStamp $transportMessageIdStamp */
+        $transportMessageIdStamp = $actualEnvelope->last(TransportMessageIdStamp::class);
+        $this->assertNotNull($transportMessageIdStamp);
+        $this->assertSame(1, $transportMessageIdStamp->getId());
```

Comment:
Notice the oddity of `1` here but `'1'` above. 

I'm a little unsure about this. In `TransportMessageIdStamp::getId()`, I add no return type... because this is truly mixed. But in `DoctrineReceivedStamp::getId()`, I currently have a `string` return type, which I adopted because  `Doctrine\DBAL\Connection::lastInsertId()` returns a string. But technically we're using an `id` column that is an integer... though the user could also ignore out table schema and create their own using a uuid (as long as the table is other wise compatible).